Managers/LuaMan.cpp

```diff
@@ -1,2829 +1,474 @@
-		m_LastError = "Script file: " + filePath + " doesn't exist!";
-		if (consoleErrors) {
-			g_ConsoleMan.PrintString("ERROR: " + m_LastError);
-			ClearErrors();
+		int error = 0;
+
+		lua_pushcfunction(m_MasterState, &AddFileAndLineToError);
+		// Load the script file's contents onto the stack and then execute it with pcall. Pcall will call the file and line error handler if there's an error by pointing 2 up the stack to it.
+		if (luaL_loadfile(m_MasterState, filePath.c_str()) || lua_pcall(m_MasterState, 0, LUA_MULTRET, -2)) {
+			// Retrieve the error message then pop it off the stack
+			m_LastError = lua_tostring(m_MasterState, -1);
+			lua_pop(m_MasterState, 1);
+			if (consoleErrors) {
+				g_ConsoleMan.PrintString("ERROR: " + m_LastError);
+				ClearErrors();
+			}
+			error = -1;
 		}
-		return -1;
-	}
+		// Pop the file and line error handler off the stack to clean it up
+		lua_pop(m_MasterState, 1);
 
-    int error = 0;
-
-    lua_pushcfunction(m_pMasterState, &AddFileAndLineToError);
-    try {
-        // Load the script file's contents onto the stack and then execute it with pcall. Pcall will call the file and line error handler if there's an error by pointing 2 up the stack to it.
-        if (luaL_loadfile(m_pMasterState, filePath.c_str()) || lua_pcall(m_pMasterState, 0, LUA_MULTRET, -2)) {
-            // Retrieve the error message then pop it off the stack
-            m_LastError = lua_tostring(m_pMasterState, -1);
-            lua_pop(m_pMasterState, 1);
-            if (consoleErrors) {
-                g_ConsoleMan.PrintString("ERROR: " + m_LastError);
-                ClearErrors();
-            }
-            error = -1;
-        }
-    } catch(const std::exception &e) {
-        m_LastError = e.what();
-        if (consoleErrors) {
-            g_ConsoleMan.PrintString("ERROR: " + m_LastError);
-            ClearErrors();
-        }
-        error = -1;
-    }
-
-    // Pop the file and line error handler off the stack to clean it up
-    lua_pop(m_pMasterState, 1);
-
-    return error;
-}
+		return error;
+	}
 
 /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-string LuaMan::GetNewPresetID()
-{
-    // Generate the new ID
-    char newID[64];
-    std::snprintf(newID, sizeof(newID), "Pre%05i", m_NextPresetID);
-    // Increment the ID so it will be diff for the next one (improve this primitive approach??)
-    m_NextPresetID++;
-
-    return string(newID);
-}
+	bool LuaMan::ExpressionIsTrue(const std::string &expression, bool consoleErrors) {
+		if (expression.empty()) {
+			return false;
+		}
+		bool result = false;
+
+		// Push the script string onto the stack so we can execute it, and then actually try to run it. Assign the result to a dedicated temp global variable.
+		if (luaL_dostring(m_MasterState, (std::string("ExpressionResult = ") + expression + std::string(";")).c_str())) {
+			// Retrieve and pop the error message off the stack
```

Comment:
I think you can get rid of this comment.

The 3 comments below can probably be combined into 1, something like:
`// Put the result of the expression on the lua stack and check its value. Need to pop it off the stack afterwards so it leaves the stack unchanged.`

In general I'm kinda on the fence about the comments in these various util methods. Most of them aren't really necessary, but the lua stack stuff is kinda esoteric, so maybe it's worth keeping them just for ease. Normally I'd say most of them should go, but they are kinda helpful just cause we don't use the lua stack stuff much.

Reply:
I'm leaving them there because lua stack stuff is dumb and generally they don't both me.